lib/ansible/cli/galaxy.py

```diff
@@ -1430,6 +1437,13 @@ def execute_list_collection(self, artifacts_manager=None):
 
                 fqcn_width, version_width = _get_collection_widths([collection])
 
+                if output_format in {'yaml', 'json'}:
+                    collections_in_paths[collection_path] = {
+                        collection.fqcn: { 'version': collection.ver }
+                    }
+
+                    continue
```

Comment:
Please move this above the `_get_collection_widths()` call to avoid running that code unnecessarily. It'd be basically the same as what's done in the else-branch that has a similar code below this snippet.

Reply:
Done